flink-ml-lib/src/main/java/org/apache/flink/ml/feature/standardscaler/StandardScalerModel.java

```diff
@@ -0,0 +1,185 @@
+                        ArrayUtils.addAll(
+                                inputTypeInfo.getFieldTypes(), TypeInformation.of(Vector.class)),
+                        ArrayUtils.addAll(inputTypeInfo.getFieldNames(), getPredictionCol()));
+
+        final String broadcastModelKey = "broadcastModelKey";
+        DataStream<StandardScalerModelData> modelDataStream =
+                StandardScalerModelData.getModelDataStream(modelDataTable);
+
+        DataStream<Row> predictionResult =
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(inputStream),
+                        Collections.singletonMap(broadcastModelKey, modelDataStream),
+                        inputList -> {
+                            DataStream inputData = inputList.get(0);
+                            return inputData.map(
+                                    new PredictOutputFunction(
+                                            broadcastModelKey,
+                                            getFeaturesCol(),
+                                            getWithMean(),
+                                            getWithStd()),
+                                    outputTypeInfo);
+                        });
+
+        return new Table[] {tEnv.fromDataStream(predictionResult)};
+    }
+
+    /** A utility function used for prediction. */
+    private static class PredictOutputFunction extends RichMapFunction<Row, Row> {
+        private final String broadcastModelKey;
+        private final String featuresCol;
+        private final boolean withMean;
+        private final boolean withStd;
+        private DenseVector mean;
+        private DenseVector scale;
+
+        public PredictOutputFunction(
+                String broadcastModelKey, String featuresCol, boolean withMean, boolean withStd) {
+            this.broadcastModelKey = broadcastModelKey;
+            this.featuresCol = featuresCol;
+            this.withMean = withMean;
+            this.withStd = withStd;
+        }
+
+        @Override
+        public Row map(Row dataPoint) {
+            if (mean == null) {
+                StandardScalerModelData modelData =
+                        (StandardScalerModelData)
+                                getRuntimeContext().getBroadcastVariable(broadcastModelKey).get(0);
+                mean = modelData.mean;
+                DenseVector std = modelData.std;
+
+                if (withStd) {
+                    scale = std;
+                    double[] scaleValues = scale.values;
+                    for (int i = 0; i < scaleValues.length; i++) {
+                        scaleValues[i] = scaleValues[i] == 0 ? 0 : 1 / scaleValues[i];
+                    }
+                }
+            }
+
+            Vector feature = ((Vector) dataPoint.getField(featuresCol)).clone();
+
+            if (withMean && feature instanceof SparseVector) {
+                feature = Vectors.toDense((SparseVector) feature);
```

Comment:
`toDense` is not used for performance here...

It is because we can only support in-place adding a `SparseVector/DenseVector` to a `DenseVecor` in `axpy`.

Reply:
Sounds good.